rag_experiment_accelerator/config/index_config.py

```diff
@@ -89,14 +89,14 @@ def __from_label_properties(cls, properties: dict) -> "IndexConfig":
             ef_construction=int(properties[IndexKey.EF_CONSTRUCTION]),
             ef_search=int(properties[IndexKey.EF_SEARCH]),
             chunking=ChunkingConfig(
-                preprocess=bool(properties[IndexKey.PREPROCESS]),
+                preprocess=bool(int(properties[IndexKey.PREPROCESS])),
                 chunk_size=int(properties[IndexKey.CHUNK_SIZE]),
                 chunking_strategy=properties[IndexKey.CHUNKING_STRATEGY],
                 overlap_size=int(properties[IndexKey.OVERLAP_SIZE]),
-                generate_title=bool(properties[IndexKey.GENERATE_TITLE]),
-                generate_summary=bool(properties[IndexKey.GENERATE_SUMMARY]),
+                generate_title=bool(int(properties[IndexKey.GENERATE_TITLE])),
+                generate_summary=bool(int(properties[IndexKey.GENERATE_SUMMARY])),
                 override_content_with_summary=bool(
```

Comment:
Severity: Major - Converting `properties[IndexKey.GENERATE_SUMMARY]` to an integer before casting to `bool` is a good practice for type safety. Ensure that `properties[IndexKey.GENERATE_SUMMARY]` is always a valid integer string to avoid `ValueError`. Consider adding validation or error handling to handle potential conversion errors.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or know how to make it better and want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.